### ISSUE TITLE
[macos] Dark mode

### ISSUE THREAD

**Issue by jerguslejko** [sharkdp/bat#641]
Hi, 

is there a way to instruct `bat` to use a different theme when the OS (macOS in particular) is using a dark theme/mode?

Right now, I use this "hack" which works fine, but I'm wondering if there is a more native approach to this.

```bash
cat() {
    bat --theme=$(defaults read -globalDomain AppleInterfaceStyle &> /dev/null && echo default || echo GitHub) $1
}
```

Thank you!

**Comment by eth-p** [comment]
I also use macOS, but I don't really feel like this is something that should be included inside bat itself. Any implementation I can think of off the top of my head is either going to be:

A. A hidden version of your hack that forks `defaults read`.

B. Require a Rust<->ObjC binding or FFI.

In the case of the latter, that's a huge dependency to add for a small feature that doesn't really add anything to improve the core functionality of bat itself.

In either case though, I can see this quickly becoming feature creep when people want to add Windows/[Insert Linux DE] support for dark mode detection. It also doesn't really take the terminal theme into consideration (e.g. I use a dark terminal with light mode), which could be counterintuitive.

But, that being said, I think adding your wrapper script to the documentation would be a great idea to help other people who want theme switching based on macOS dark/light mode.

**Comment by sharkdp** [comment]
@jerguslejko Thank you for your feedback!

I'm with @eth-p on this.

> But, that being said, I think adding your wrapper script to the documentation would be a great idea to help other people who want theme switching based on macOS dark/light mode.

:+1:

**Comment by jerguslejko** [comment]
@eth-p @sharkdp makes total sense, thanks! I can submit a PR with the wrapper script. Should that be included in the "Customization" section?

**Comment by sharkdp** [comment]
> I can submit a PR with the wrapper script

Sounds great. Maybe in a new `scripts` folder in `/assets`?

> Should that be included in the "Customization" section?

:+1:

**Comment by sharkdp** [comment]
Closed in #662 by @jerguslejko

**Comment by aykevl** [comment]
> In either case though, I can see this quickly becoming feature creep when people want to add Windows/[Insert Linux DE] support for dark mode detection. It also doesn't really take the terminal theme into consideration (e.g. I use a dark terminal with light mode), which could be counterintuitive.

Just for anybody coming across this issue: this isn't actually necessary. See #1746 for details.
